modules/MatchProvider.js

```diff
@@ -17,34 +18,35 @@ class MatchProvider extends React.Component {
     // **IMPORTANT** we must mutate matches, never reassign, in order for
     // server rendering to work w/ the two-pass render approach for Miss
     this.matches = []
```

Comment:
Hmm yeah looks like `hasMatches` is never set again on the server, because it is only ever set in `<MatchProvider>`'s `componentDidUpdate` and `componentDidMount`, which don't get called on the server. I'll put a PR in for this if I have time.